alibi_detect/utils/pytorch/kernels.py

```diff
@@ -20,21 +56,307 @@ def sigma_median(x: torch.Tensor, y: torch.Tensor, dist: torch.Tensor) -> torch.
 
     Returns
     -------
-    The computed bandwidth, `sigma`.
+    The logrithm of the computed bandwidth, `log-sigma`.
     """
     n = min(x.shape[0], y.shape[0])
     n = n if (x[:n] == y[:n]).all() and x.shape == y.shape else 0
     n_median = n + (np.prod(dist.shape) - n) // 2 - 1
     sigma = (.5 * dist.flatten().sort().values[n_median].unsqueeze(dim=-1)) ** .5
-    return sigma
+    return sigma.log()
 
 
-class GaussianRBF(nn.Module):
+class KernelParameter:
     def __init__(
         self,
-        sigma: Optional[torch.Tensor] = None,
-        init_sigma_fn: Callable = sigma_median,
-        trainable: bool = False
+        value: torch.Tensor = None,
+        init_fn: Optional[Callable] = None,
+        requires_grad: bool = False,
+        requires_init: bool = False
+    ) -> None:
+        """
+        Parameter class for kernels.
+
+        Parameters
+        ----------
+        value
+            The pre-specified value of the parameter.
+        init_fn
+            The function used to initialize the parameter.
+        requires_grad
+            Whether the parameter requires gradient.
+        requires_init
+            Whether the parameter requires initialization.
+        """
+        super().__init__()
+        self.value = nn.Parameter(value if value is not None else torch.ones(1),
+                                  requires_grad=requires_grad)
+        self.init_fn = init_fn
+        self.requires_init = requires_init
+
+
+class BaseKernel(nn.Module):
+    def __init__(self, active_dims: list = None, feature_axis: int = -1) -> None:
```

Comment:
As mentioned by @ojcobb [here](https://github.com/SeldonIO/alibi-detect/pull/558#issuecomment-1208162385), it is unclear why both `active_dims` and `feature_axis` are present. The parameter description also doesn't clarify this and should be more descriptive.

Reply:
What would be a concrete example of where `feature_axis != -1`?

Reply:
I thought if we use an image batch with shape`[n_batch, n_pixel, n_colour]`, if we want to select a sub-patch of every image across all images, we might want `feature_axis=1` and 'active_dims=[...]'.